services/workflows-service/src/notion.service.ts

```diff
@@ -0,0 +1,113 @@
+import { AppLoggerService } from '@/common/app-logger/app-logger.service';
+import { ConfigService } from '@nestjs/config';
+import { Client } from '@notionhq/client';
+import {
+  PageObjectResponse,
+  QueryDatabaseResponse,
+} from '@notionhq/client/build/src/api-endpoints';
+import { z } from 'zod';
+import { env } from '@/env';
+import { Injectable } from '@nestjs/common';
+
+@Injectable()
+export class NotionService {
+  private readonly client: Client;
+
+  constructor(
+    private readonly configService: ConfigService,
+    private readonly logger: AppLoggerService,
+  ) {
+    if (!env.NOTION_API_KEY) {
```

Comment:
Is it make sense to allow this variable as optional and once the service is initialized you will throw an exception?

Reply:
My solution is to create a provider that dynamically selects the appropriate service based on the configuration

something like:
```
export const dynamicServiceProvider: Provider = {
  provide: MyService,
  useFactory: (configService: ConfigService) => {
    const key = configService.get<string>('NOTION_KEY');
    if (key) {
      return new NotionService();
    }
    return new MockNotionService();
  },
  inject: [ConfigService],
};
```

MockNotionService can throw error when a concrete service api is called

Reply:
I prefer it to be required so in production if you will forget to set the env var it won't provision the pod. The alternative you suggested will notify only when a user tries to use the feature.